batch/Makefile

```diff
@@ -40,14 +40,60 @@ deploy: build
 	kubectl -n $(NAMESPACE) apply -f service-account.yaml.out
 	kubectl -n $(NAMESPACE) apply -f deployment.yaml.out
 
-.PHONY: create-build-worker-image-instance
-create-build-worker-image-instance:
+.PHONY: gcp-create-build-worker-image-instance
+gcp-create-build-worker-image-instance:
 	-gcloud -q compute --project $(PROJECT) instances delete --zone=$(ZONE) build-batch-worker-image
-	python3 ../ci/jinja2_render.py '{"global":{"docker_root_image":"$(DOCKER_ROOT_IMAGE)"}}' build-batch-worker-image-startup.sh build-batch-worker-image-startup.sh.out
-	gcloud -q compute --project $(PROJECT) instances create --zone=$(ZONE) build-batch-worker-image --machine-type=n1-standard-1 --network=default --network-tier=PREMIUM --metadata-from-file startup-script=build-batch-worker-image-startup.sh.out --no-restart-on-failure --maintenance-policy=MIGRATE --scopes=https://www.googleapis.com/auth/cloud-platform --image=$$(gcloud compute images list --standard-images --filter 'family="ubuntu-minimal-2004-lts"' --format='value(name)') --image-project=ubuntu-os-cloud --boot-disk-size=10GB --boot-disk-type=pd-ssd
+	python3 ../ci/jinja2_render.py '{"global":{"docker_root_image":"$(DOCKER_ROOT_IMAGE)"}}' build-batch-worker-image-startup-gcp.sh build-batch-worker-image-startup-gcp.sh.out
+	gcloud -q compute --project $(PROJECT) instances create --zone=$(ZONE) build-batch-worker-image --machine-type=n1-standard-1 --network=default --network-tier=PREMIUM --metadata-from-file startup-script=build-batch-worker-image-startup-gcp.sh.out --no-restart-on-failure --maintenance-policy=MIGRATE --scopes=https://www.googleapis.com/auth/cloud-platform --image=$$(gcloud compute images list --standard-images --filter 'family="ubuntu-minimal-2004-lts"' --format='value(name)') --image-project=ubuntu-os-cloud --boot-disk-size=10GB --boot-disk-type=pd-ssd
 
-.PHONY: create-worker-image
-create-worker-image:
+.PHONY: gcp-create-worker-image
+gcp-create-worker-image:
 	-gcloud -q compute --project $(PROJECT) images delete batch-worker-12
 	gcloud -q compute --project $(PROJECT) images create batch-worker-12 --source-disk=build-batch-worker-image --source-disk-zone=$(ZONE)
 	gcloud -q compute --project $(PROJECT) instances delete --zone=$(ZONE) build-batch-worker-image
+
+.PHONY: az-create-build-worker-image-instance
+az-create-build-worker-image-instance: export BATCH_WORKER_PRINCIPAL_ID := $(shell az identity show --resource-group $(RESOURCE_GROUP) --name batch-worker --query principalId --output tsv)
+az-create-build-worker-image-instance: export BATCH_WORKER_IDENTITY := $(shell az identity show --resource-group $(RESOURCE_GROUP) --name batch-worker --query id --output tsv)
```

Comment:
I spent a long time trying to figure out how to do this properly. I'm not actually sure it's the correct way. I tried a bunch of things. The issue about putting the definition of the environment variable in the actual rule is each command is run in it's own subshell and not all together in the same shell. So I'd have to define the variable at the beginning of each command which was difficult to read. We could inline the commands instead of using environment variables, but that was also difficult to read.

https://stackoverflow.com/questions/23843106/how-to-set-child-process-environment-variable-in-makefile

@danking Can you comment on whether there's a better way to define environment variables at runtime that can then be used by multiple commands in the rule?